examples/ex-batched-matrix-multiply.cpp

```diff
@@ -27,264 +27,469 @@
-//Indexing is equivalent to A(e,r,c) A[c + NCOLS*(r + NROWS*e)]
-// e - 0th index
-// r - 1st index
-// c - 2nd index
-  auto layout = RAJA::make_permuted_layout({{NMAT, NROWS, NCOLS}}, RAJA::as_array<RAJA::Perm<0,1,2> >::get() );
-  
-//RAJA::Layout is templated on dimension, argument type, and index with unit stride (in this case argument 2 has unit stride)
-  RAJA::View<double, RAJA::Layout<3, Index_type, 2> > Aview(A,layout);
-  RAJA::View<double, RAJA::Layout<3, Index_type, 2> > Bview(B,layout);
-  RAJA::View<double, RAJA::Layout<3, Index_type, 2> > Cview(C,layout);
+  //
+  // Layout 1
+  //
+  // make_permuted_layout takes the number of entries in each dimension and an
+  // array indicating slowest to fastest stride
+  // Indexing is equivalent to A(e,r,c) A[c + NCOLS*(r + NROWS*e)]
+  auto layout =
+      RAJA::make_permuted_layout({{NMAT, NROWS, NCOLS}},
+                                 RAJA::as_array<RAJA::Perm<0, 1, 2> >::get());
 
-//
-//Allocate space for data in layout 2
-//
-  double * Al2 = memoryManager::allocate<double>(NCOLS*NROWS*NMAT);
-  double * Bl2 = memoryManager::allocate<double>(NCOLS*NROWS*NMAT);
-  double * Cl2 = memoryManager::allocate<double>(NCOLS*NROWS*NMAT);
-  
-//
-//Permuted layout - equivalent to indexing via 
-//A[e + NELEM*(r + NROWS*e)]
-// r - 1st index
-// c - 2nd index
-// e - 0th index
-  auto layout2 = RAJA::make_permuted_layout({{NMAT, NROWS, NCOLS}}, RAJA::as_array<RAJA::Perm<1,2,0> >::get() );
-  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Al2view(Al2,layout2);
-  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Bl2view(Bl2,layout2);
-  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Cl2view(Cl2,layout2);
+  // RAJA::Layout is templated on dimension, argument type, and index with unit
+  // stride (in this case argument 2 has unit stride)
+  RAJA::View<double, RAJA::Layout<3, Index_type, 2> > Aview(A, layout);
+  RAJA::View<double, RAJA::Layout<3, Index_type, 2> > Bview(B, layout);
+  RAJA::View<double, RAJA::Layout<3, Index_type, 2> > Cview(C, layout);
 
-//
-//Initialize data
-//
-  for(Index_type e=0; e<NMAT; ++e){
-    for(Index_type row=0; row<NROWS; ++row){
-      for(Index_type col=0; col<NCOLS; ++col){
-        Aview(e,row,col) = rand() % 50 + 1; 
-        Bview(e,row,col) = rand() % 50 + 1; 
-        Al2view(e,row,col) = Aview(e,row,col);
-        Bl2view(e,row,col) = Bview(e,row,col);
+  //
+  // Allocate space for data in layout 2
+  //
+  double *Al2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+  double *Bl2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+  double *Cl2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+
+  //
+  // Permuted layout - equivalent to indexing via
+  // A[e + NELEM*(r + NROWS*c)]
+  auto layout2 =
+      RAJA::make_permuted_layout({{NMAT, NROWS, NCOLS}},
+                                 RAJA::as_array<RAJA::Perm<1, 2, 0> >::get());
+  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Al2view(Al2, layout2);
```

Comment:
It may be easier to read if you change these to 'Aview2', etc. What is the 'l' for?